### PR TITLE
add fallback tokenizer

### PR DESCRIPTION
**Pull request by JerryKwan**
add fallback tonenizer if tiktoken can not get encoding from model name 
support llm services which provide openai compatibility api such as ollama

### CODE REVIEW

**Comment by lapp0** [comment]
Doesn't `cl100k_base` only apply to gpt4 and gpt3? Or am I missing something?

**Comment by JerryKwan** [comment]
@lapp0 
As far as I know, gpt-4, gpt-3.5-turbo, text-embedding-ada-002, text-embedding-3-small, text-embedding-3-large use cl100k_base, and GPT-3 models like davinci use r50k_base

File: outlines/models/openai.py [hunk 1]
```diff
@@ -439,7 +439,11 @@ def openai_model(
         config = OpenAIConfig(model=model_name)
 
     client = AsyncOpenAI(**openai_client_params)
-    tokenizer = tiktoken.encoding_for_model(model_name)
+
+    try:
+        tokenizer = tiktoken.encoding_for_model(model_name)
+    except KeyError:
```
Comment (rlouf) [inline outlines/models/openai.py:445]: raise a warning so the user knows we made a choice for them, and what this choice is

**Comment by rlouf** [comment]
Please run `pre-commit` locally and push the changes

**Review by lapp0 — APPROVED** [review]
Thanks!

**Pull request merged**